experimenter/experimenter/experiments/changelog_utils.py

```diff
@@ -87,3 +92,66 @@ def generate_nimbus_changelog(experiment, changed_by, message, changed_on=None):
         message=message,
         changed_on=changed_on,
     )
+
+
+def get_formatted_change_object(field_name, field_diff, changelog, timestamp):
+    event_name = ChangeEventType.GENERAL.name
+    field_instance = NimbusExperiment._meta.get_field(field_name)
+    field_display_name = (
+        field_instance.verbose_name
+        if hasattr(field_instance, "verbose_name")
+        else field_name
+    )
+
+    old_value = field_diff["old_value"]
+    new_value = field_diff["new_value"]
+
+    if isinstance(
+        field_instance,
+        (models.ManyToManyField, models.OneToOneField, models.ManyToOneRel),
+    ):
+        event_name = ChangeEventType.DETAILED.name
+        if field_name in RelationalFields.NATIVE_MODELS:
+            field_model = field_instance.related_model
+            data = field_model.objects.all()
+            values = list(data.filter(pk__in=old_value).values_list("name", flat=True))
+            old_value = values if values else old_value
+            values = list(data.filter(pk__in=new_value).values_list("name", flat=True))
+            new_value = values if values else new_value
+
+        old_value = json.dumps(old_value, indent=2)
+        new_value = json.dumps(new_value, indent=2)
+
+    if isinstance(field_instance, (models.JSONField, ArrayField)):
+        event_name = ChangeEventType.DETAILED.name
+        if old_value is not None:
+            old_value = json.dumps(old_value, indent=2)
+        if new_value is not None:
+            new_value = json.dumps(new_value, indent=2)
+
+    if isinstance(field_instance, models.BooleanField):
+        event_name = ChangeEventType.BOOLEAN.name
+
+    if field_name == "status" or field_name == "publish_status":
+        event_name = ChangeEventType.STATE.name
+
+    if event_name == ChangeEventType.DETAILED.name:
+        change_message = f"{changelog.changed_by} changed value of {field_display_name}"
+    elif event_name == ChangeEventType.BOOLEAN.name:
+        change_message = (
+            f"{changelog.changed_by} set the {field_display_name} as {new_value}"
+        )
+    else:
+        change_message = (
+            f"{changelog.changed_by} changed value of {field_display_name} from "
+            f"{old_value} to {new_value}"
+        )
```

Comment:
Why do we need to specialize the boolean case?

Reply:
for rendering a different change message for change of a binary type (true/false) as it doesn't make sense to put the old value and the new value in the message if they arent going to be different